File: atv8/exerc8.go
package main
import "fmt"

type Empregado struct {
    Nome   string
    Cargo  string
    Salario float64
}

type Empresa struct {
    Nome      string
    Empregados []Empregado
}

func (e *Empresa) adicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}

func (e Empresa) listarEmpregados() {
    fmt.Println("Empregados da empresa", e.Nome + ":")
    for _, empregado := range e.Empregados {
        fmt.Println(empregado.Nome, "-", empregado.Cargo, "-", empregado.Salario)
    }
}

func main() {
    empresa := Empresa{Nome: "Tech Corp"}
    e1 := Empregado{"João", "Desenvolvedor", 3000.0}
    e2 := Empregado{"Maria", "Designer", 3500.0}

    empresa.adicionarEmpregado(e1)
    empresa.adicionarEmpregado(e2)

    empresa.listarEmpregados()
}

File: atv8/exerc8.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def listar_empregados(self):
        print(f"Empregados da empresa {self.nome}:")
        for empregado in self.empregados:
            print(f"{empregado.nome} - {empregado.cargo} - R${empregado.salario}")

empresa = Empresa("Tech Corp")
e1 = Empregado("João", "Desenvolvedor", 3000.0)
e2 = Empregado("Maria", "Designer", 3500.0)

empresa.adicionar_empregado(e1)
empresa.adicionar_empregado(e2)

empresa.listar_empregados()
